setupdune.sh

```diff
@@ -24,8 +45,22 @@ install-dune() {
   (set -x; dune --version)
 }
 
-lock() {
-  (set -x; cd "$SETUPDUNEDIR" && dune pkg lock)
+enable_pkg() {
+  case "$(dune --version)" in
+    3.19*|3.20*)
+      (set -x; cd "$SETUPDUNEDIR" && test -d dune.lock) || dune_aux pkg lock
+      ;;
+    *)
+      CONFIG_DIR="${XDG_CONFIG_HOME:-$HOME/.config}/dune"
+      mkdir -p "$CONFIG_DIR"
+      if test -e "$CONFIG_DIR/config"; then
+        printf '::warning::%s already exists, assume it enables package management\n' "$CONFIG_DIR/config" >&2
```

Comment:
You could call `dune pkg enabled` here to check, if that is false then you could terminate.